cli/command/registry/login.go

```diff
@@ -143,7 +143,8 @@ func runLogin(dockerCli command.Cli, opts loginOptions) error { //nolint: gocycl
 	creds := dockerCli.ConfigFile().GetCredentialsStore(serverAddress)
 
 	store, isDefault := creds.(isFileStore)
-	if isDefault {
+	// Display a warning if we're storing the users password (not a token)
+	if isDefault && authConfig.IdentityToken == "" {
```

Comment:
Perhaps better to check for a non-empty password instead of relying on the side-effect of an IdentityToken clearing the password;

```suggestion
	if isDefault && authConfig.Password != "" {
```

Reply:
Good suggestion - fixed.